### PR TITLE
Reader: Use localize() instead of i18n mixin

### PR DESCRIPTION
**Pull request by ockham**
Let's continue the tradition of using the Reader as a guinea pig for trying out new codemods 😁 

Branch made by running the codemod from https://github.com/Automattic/wp-calypso/pull/13597, and `s/localize\((\S*)\)/localize( $1 )/g`.

To test:
* Verify that the Reader is still properly localized.

### CODE REVIEW

**Comment by samouri** [comment]
Any reason the codemod is wrapping the declaration instead of the line with the export (how we usually do it)

**Comment by gziolo** [comment]
I was about to ask about the same :) It might break one of the Babel's transformation which gives the name to the components. Another thing is that `React.createClass` is something that Eslint no longer loves :)

**Comment by samouri** [comment]
> Let's continue the tradition of using the Reader as a guinea pig for trying out new codemods 😁

Love it btw 🎉

**Comment by ockham** [comment]
> Any reason the codemod is wrapping the declaration instead of the line with the export (how we usually do it)?

It was way easier to wrap `localize` directly around `createClass` (for which there is a readily available util in `react-codemod`) than to account for different possible cases such as `export default`, non-default `export`, variable assignment, etc.

**Comment by ockham** [comment]
> Any reason the codemod is wrapping the declaration instead of the line with the export (how we usually do it)?

Thanks to @jsnajdr's contribution to #13597, the codemod is now smarter! I re-ran it on `client/reader` and updated this PR accordingly.

**Comment by gziolo** [comment]
This makes linter very unhappy. Errors from the first container:

```

/home/ubuntu/wp-calypso/client/reader/update-notice/index.jsx
  39:2  error  Mixed spaces and tabs                           no-mixed-spaces-and-tabs
  42:1  error  Line 42 exceeds the maximum line length of 140  max-len

/home/ubuntu/wp-calypso/client/reader/following-edit/sort-controls.jsx
  33:2  error  Mixed spaces and tabs  no-mixed-spaces-and-tabs

/home/ubuntu/wp-calypso/client/reader/recommendations/posts/index.jsx
  18:3  error  Unexpected var, use let or const instead  no-var
  22:2  error  Mixed spaces and tabs                     no-mixed-spaces-and-tabs

/home/ubuntu/wp-calypso/client/reader/stream/empty.jsx
  40:2  error  Mixed spaces and tabs  no-mixed-spaces-and-tabs

/home/ubuntu/wp-calypso/client/reader/following-edit/import-button.jsx
  72:2  error  Mixed spaces and tabs  no-mixed-spaces-and-tabs

✖ 7 problems (7 errors, 0 warnings)
```

There are 3 more container to check, but it could be easily fixed by prettier (see #12260) :)

Expect me to advertise `prettier` more each time we introduce new codemod :D

**Comment by jsnajdr** [comment]
> This makes linter very unhappy. Errors from the first container:

Some `eslint --fix` processing will be necessary here.
- when formatting the output, recast doesn't support the `space-in-parens` rule (or maybe it's just not turned on?)
- it has a bug when formatting JSX -- inserts spaces before the opening tag despite the `useTabs` option

**Comment by ockham** [comment]
I ran `./node_modules/.bin/eslint --fix client/reader/`, and it doesn't seem to be able to fix `no-mixed-spaces-and-tabs` (yet -- I might need to check the Changelog and maybe update to a more recent version).

**Comment by ockham** [comment]
@gziolo Anyway, it looks like most of these lints aren't even introduced by the codemod but seem to pre-exist, no? (Check the affected lines.) If that's the case, I'd say it's fine to ignore them and consider them an orthogonal issue.

**Comment by ockham** [comment]
Hmm, on second glance, I might be wrong about that 😳

**Comment by ockham** [comment]
> I ran `./node_modules/.bin/eslint --fix client/reader/`, and it doesn't seem to be able to fix `no-mixed-spaces-and-tabs` (yet -- I might need to check the Changelog and maybe update to a more recent version).

Computer says no: https://github.com/eslint/eslint/issues/6873

**Comment by ockham** [comment]
Resorted to using Atom to replace `	    ` with `		`. Which seems to have worked 😄

**Comment by gziolo** [comment]
IDE is your friend. That works, too :)

File: client/reader/liked-stream/main.jsx [hunk 1]
```diff
@@ -3,6 +3,8 @@
  */
 import React from 'react';
```
Comment (gziolo) [inline client/reader/liked-stream/main.jsx]: I'm wondering why it adds empty line here and in couple of other places?

**Comment by jsnajdr** [comment]
> I ran ./node_modules/.bin/eslint --fix client/reader/, and it doesn't seem to be able to fix no-mixed-spaces-and-tabs

If you mean the spaces added before an opening JSX element, I isolated the bug in `recast` and will be coming up with a patch. It won't be a problem for long, hopefully.

File: client/reader/site-stream/featured.jsx [hunk 2]
```diff
@@ -19,7 +20,7 @@ import {
 } from 'reader/stats';
 import cssSafeUrl from 'lib/css-safe-url';
 
-export default React.createClass( {
+export default localize(React.createClass({
```
Comment (gziolo) [inline client/reader/site-stream/featured.jsx]: Missing spaces inside `localize`.

**Comment by gziolo** [comment]
I had two minor comments, otherwise it looks neat. I think it needs to be tested by someone from Reader team and we are good to 🚢

File: client/reader/liked-stream/main.jsx (hunk 1 above)
Comment (ockham) [inline client/reader/liked-stream/main.jsx]: I think that's https://github.com/benjamn/recast/issues/371

**Comment by ockham** [comment]
>> I ran ./node_modules/.bin/eslint --fix client/reader/, and it doesn't seem to be able to fix no-mixed-spaces-and-tabs
>
>If you mean the spaces added before an opening JSX element, I isolated the bug in `recast` and will be coming up with a patch. It won't be a problem for long, hopefully.

Nice! Wanna add a link to the PR for reference? 😄

**Comment by ockham** [comment]
> I think it needs to be tested by someone from Reader team and we are good to 🚢 

@blowery @bluefuton @samouri pls? 😄

**Comment by jsnajdr** [comment]
> Nice! Wanna add a link to the PR for reference?
Here it is:
https://github.com/benjamn/recast/pull/404

**Comment by samouri** [comment]
I'll give this a go 🤞

**Comment by samouri** [comment]
I'm on the rest of the style errors *ugh*

**Comment by ockham** [comment]
Thanks for merging @samouri!

**Pull request merged**